compiler/rustc_session/src/session.rs

```diff
@@ -1092,20 +1168,31 @@ impl Session {
         || self.opts.unstable_opts.sanitizer.intersects(SanitizerSet::ADDRESS | SanitizerSet::MEMORY | SanitizerSet::HWADDRESS)
     }
 
+    // JUSTIFICATION: defn of the suggested wrapper fn
+    #[cfg_attr(not(bootstrap), allow(rustc::bad_opt_access))]
     pub fn link_dead_code(&self) -> bool {
         self.opts.cg.link_dead_code.unwrap_or(false)
     }
 
+    // JUSTIFICATION: defn of the suggested wrapper fn
+    #[cfg_attr(not(bootstrap), allow(rustc::bad_opt_access))]
     pub fn instrument_coverage(&self) -> bool {
-        self.opts.instrument_coverage()
+        self.opts.cg.instrument_coverage.unwrap_or(InstrumentCoverage::Off)
```

Comment:
maybe put the `self.opts.cg.instrument_coverage.unwrap_or(InstrumentCoverage::Off)` as a method of `CodegenOptions` or as a new private function? :thinking:

Reply:
Fixed